sdk/core/core/inc/az_span.h

```diff
@@ -198,24 +157,24 @@ AZ_NODISCARD az_span az_span_from_str(char* str);
 AZ_NODISCARD az_span az_span_slice(az_span span, int32_t start_index, int32_t end_index);
```

Comment:
The docs should mention what passing -1 for end_index does

Reply:
Ah yes thanks for this. Was going to suggest it when I was using it in IoT APIs

Reply:
I prefer a named method for this behavior (`az_span_slice_to_end` OR `az_span_slice_from_start`) and would like to make this change instead/remove the -1 sentinel. To me, that is easier to understand, the method names become self-documenting, and then we can remove the check for -1.

If we don't want that, then I will update the docs to mention -1.